applications/GeoMechanicsApplication/tests/test_prescribed_derivatives.py

```diff
@@ -0,0 +1,101 @@
+# from KratosMultiphysics import * as Kratos
+
+import sys
+import os
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.GeoMechanicsApplication as KratosGeo
+import KratosMultiphysics.GeoMechanicsApplication.geomechanics_analysis as analysis
+
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+import test_helper
+
+
+class KratosGeoMechanicsPrescribedDerivatives(KratosUnittest.TestCase):
+    """
+    This class contains tests which check if prescribed derivatives are correctly applied to the model and not overwritten
+    """
+
+    def setUp(self):
+        # Code here will be placed BEFORE every test in this TestCase.
+        pass
+
+    def tearDown(self):
+        # Code here will be placed AFTER every test in this TestCase.
+        pass
+
+    def test_prescribed_acceleration(self):
+        """
+        This test checks if the prescribed acceleration in the x-direction is correctly applied to the model
+        :return:
+        """
+
+        test_name = 'prescribed_acceleration'
+        file_path = test_helper.get_file_path(os.path.join('.', 'prescribed_derivative_tests', test_name))
+        test_helper.run_kratos(file_path)
+
+        output_file_path = os.path.join(file_path, test_name + '.post.res')
+        output_reader = test_helper.GiDOutputFileReader()
+        output_data = output_reader.read_output_from(output_file_path)
+
+        times = [0.05, 0.1, 0.15, 0.2]
+        expected_x_accelerations = [-0.01, -0.005, -0.003, 0.01]
+        for idx, time in enumerate(times):
+            accelerations = test_helper.GiDOutputFileReader.nodal_values_at_time("ACCELERATION", time, output_data,
+                                                                                 node_ids=[1, 2, 3])
+            for acceleration in accelerations:
+                self.assertAlmostEqual(expected_x_accelerations[idx], acceleration[0], 6)
+                self.assertAlmostEqual(0.0, acceleration[1], 6)
+                self.assertAlmostEqual(0.0, acceleration[2], 6)
```

Comment:
Rather than using an index (`idx`), I would prefer to iterate over a ["zipped" range](https://docs.python.org/3/library/functions.html#zip):
```suggestion
        for expected_x_acceleration, time in zip(expected_x_accelerations, times):
            accelerations = test_helper.GiDOutputFileReader.nodal_values_at_time("ACCELERATION", time, output_data,
                                                                                 node_ids=[1, 2, 3])
            for acceleration in accelerations:
                self.assertAlmostEqual(expected_x_acceleration, acceleration[0], 6)
                self.assertAlmostEqual(0.0, acceleration[1], 6)
                self.assertAlmostEqual(0.0, acceleration[2], 6)
```

Similar changes are needed for the next two test (for velocity and "dt_water_pressure").

One more remark: I see that we have a nested `for` loop. I would, however, expect one and only one vector per time step. Shouldn't we check for that as well? I mean, what happens if no output result was calculated and `accelerations` is empty? Wouldn't the test still pass?

Reply:
Nice, didn't know about this zipping functionality. I added a check to check we get 3 vectors per time step (1 for each node)